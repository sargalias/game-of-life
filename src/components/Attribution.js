import React from 'react';

const Attribution = (props) => (
  <div id="attribution">
    <p id="author">
      By <a href={props.authorUrl} target="_blank">{props.authorName}</a>
    </p>
    <p id="code-repository">
      View code on <a href={props.codeRepositoryUrl} target="_blank">GitHub</a>
    </p>
  </div>
);

export default Attribution;