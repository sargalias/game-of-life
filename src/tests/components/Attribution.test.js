import React from 'react';
import { shallow } from 'enzyme';
import Attribution from '../../components/Attribution';


test('Attribution should render correctly', () => {
  const wrapper = shallow(<Attribution
    authorName="testName"
    authorUrl="https://test.com"
    codeRepositoryUrl="https://test-code.com"
  />);
  expect(wrapper).toMatchSnapshot();
});

test('Attribution should correctly use its authorName prop', () => {
  const authorName = 'test name';
  const wrapper = shallow(<Attribution authorName={authorName} />);
  expect(wrapper.find('#author').text()).toContain(authorName);
});

test('Attribution should correctly use its authorUrl prop', () => {
  const authorUrl = 'https://test.com';
  const wrapper = shallow(<Attribution authorUrl={authorUrl} />);
  expect(wrapper.find('#author a').render().attr('href')).toBe('https://test.com');
});

test('Attribution should correctly use its codeRepositoryUrl prop', () => {
  const codeRepositoryUrl = 'https://test.com';
  const wrapper = shallow(<Attribution codeRepositoryUrl={codeRepositoryUrl} />);
  expect(wrapper.find('#code-repository a').render().attr('href')).toBe(codeRepositoryUrl);
});
