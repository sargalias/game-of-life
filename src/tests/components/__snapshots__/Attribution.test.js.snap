// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Attribution should render correctly 1`] = `
<div
  id="attribution"
>
  <p
    id="author"
  >
    By 
    <a
      href="https://test.com"
      target="_blank"
    >
      testName
    </a>
  </p>
  <p
    id="code-repository"
  >
    View code on 
    <a
      href="https://test-code.com"
      target="_blank"
    >
      GitHub
    </a>
  </p>
</div>
`;
